app/services/PaymentAPIService.scala

```diff
@@ -65,48 +56,24 @@ class PaymentAPIService(wsClient: WSClient, paymentAPIUrl: String) {
       .execute()
   }
 
-  def decodePaymentResponse(response: WSResponse): Option[PaymentApiError] = {
-    decode[PaymentApiError](response.body).fold(
+  def decodePaymentResponse(response: WSResponse): Option[PaymentAPIResponse[PayPalError, PaypalExecuteSuccessResponse]] = {
+    decode[PaymentAPIResponse[PayPalError, PaypalExecuteSuccessResponse]](response.body).fold(
       failure => {
-        SafeLogger.error(scrub"Unable to decode PaymentAPIError: ${response.body}. Message is ${failure.getMessage}")
+        SafeLogger.error(scrub"Unable to decode payment API response: ${response.body}. Message is ${failure.getMessage}")
         None
       },
       resp => {
+        resp.error.foreach(logErrorResponse)
         Some(resp)
       }
     )
   }
 
-  def logErrorResponse(paymentApiError: Option[PaymentApiError]): Unit = {
-    paymentApiError.foreach(err => {
-      if (err.error.errorName.contains("INSTRUMENT_DECLINED")) {
-        SafeLogger.info("Paypal payment failed with 'INSTRUMENT_DECLINED' response.")
-      } else {
-        SafeLogger.error(scrub"Paypal payment failed due to ${err.error.errorName} error. Full message: ${err.error.message}")
-      }
-    })
-  }
-
-  def isDuplicatePaymentResponse(errorOpt: Option[PaymentApiError]): Boolean = {
```

Comment:
Has this been removed?

Reply:
the logic has been moved to the `processPaymentApiResponse` function